cirq-core/cirq/ops/pauli_string.py

```diff
@@ -1159,11 +1159,33 @@ def __mul__(self, other):
             return self._as_pauli_string() * other._as_pauli_string()
         if isinstance(other, (PauliString, complex, float, int)):
             return self._as_pauli_string() * other
+        if isinstance(other, gate_operation.GateOperation):
+            # if multiplication by an exponentiated pauli
+            # solve it, if exponent is integer
+            if (
+                isinstance(
+                    other.gate,
+                    (common_gates.XPowGate, common_gates.YPowGate, common_gates.ZPowGate),
+                )
+                and len(other.qubits) == 1
+            ):
+                power = other.gate.exponent
+                if power % 2 == 0:
+                    return self
+                elif power % 2 == 1:
+                    gates = {
+                        common_gates.XPowGate: pauli_gates.X,
+                        common_gates.YPowGate: pauli_gates.Y,
+                        common_gates.ZPowGate: pauli_gates.Z,
+                    }
+                    return self * gates[type(other.gate)].on(other.qubits[0])
         return NotImplemented
 
     def __rmul__(self, other):
         if isinstance(other, (PauliString, complex, float, int)):
             return other * self._as_pauli_string()
+        if isinstance(other, gate_operation.GateOperation):
+            return self * other
```

Comment:
this assumes that the gates commute ... which is not always true. the tests pass because the paulis act on two different qubits so they commute... but if they act on the same qubit then they anticommute

Reply:
Ups, I've fixed the code and the tests to catch the error